lib/ProductOpener/Display.pm

```diff
@@ -1316,6 +1318,15 @@ sub display_date_iso($t) {
 	}
 }
 
+
+=head2 display_error ( $error_message, $status )
+
+Display an error message using the site template.
+
+The request is not terminated by this function, it will continue to run.
+
+=cut
+
 sub display_error($error_message, $status) {
```

Comment:
we should take care of the context.
An API request should return a json instead.
I would use a parameter for that, but also have a function is_api_request($request_ref) to determine that.

But maybe this is a separate PR.

Reply:
> An API request should return a json instead.

Agreed, but let's make it in another PR, as there may be side effects, and we need to discuss what the right format of that JSON is with the API users.

Reply:
agree, opened #7365